src/main/java/net/snowflake/client/jdbc/DefaultConnectionHandler.java

```diff
@@ -0,0 +1,179 @@
+package net.snowflake.client.jdbc;
+
+import static net.snowflake.client.core.SessionUtil.CLIENT_SFSQL;
+import static net.snowflake.client.core.SessionUtil.JVM_PARAMS_TO_PARAMS;
+import static net.snowflake.client.jdbc.SnowflakeUtil.systemGetProperty;
+
+import java.sql.*;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.client.core.*;
+import net.snowflake.client.jdbc.telemetryOOB.TelemetryService;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+import net.snowflake.common.core.LoginInfoDTO;
+
+/**
+ * The default ConnectionHandler used by SnowflakeConnectionV(x). Unless a separate implementation
+ * is provided, a DefaultConnectionHandler will be constructed automatically by the Connection
+ * class.
+ */
+public class DefaultConnectionHandler implements ConnectionHandler {
+
+  private static final SFLogger logger = SFLoggerFactory.getLogger(DefaultConnectionHandler.class);
+
+  private final SFSession sfSession;
+  private final SnowflakeConnectString conStr;
+  private final boolean skipOpen;
+
+  /**
+   * Constructs a DefaultConnectionHandler using a SnowflakeConnectString. This can be done by using
+   * SnowflakeConnectString.parse(url, info), where url is a connection url and info is a
+   * java.util.Properties
+   *
+   * @param conStr A SnowflakeConnectString object
+   */
+  public DefaultConnectionHandler(SnowflakeConnectString conStr) {
+    this(conStr, false);
+  }
+
+  /**
+   * Constructs a DefaultConnectionHandler using a SnowflakeConnectString. This can be done by using
+   * SnowflakeConnectString.parse(url, info), where url is a connection url and info is a
+   * java.util.Properties
+   *
+   * @param conStr A SnowflakeConnectString object
+   */
+  public DefaultConnectionHandler(SnowflakeConnectString conStr, boolean skipOpen) {
```

Comment:
Why would we need to skip session.open()?

Reply:
I think I added that to support a test that would try to create a SnowflakeConnectionV1 without actually opening a connection. See `fakeConnection` in a SnowflakeConnectionV1 constructor.